warmup/summary_statistics.py

```diff
@@ -136,6 +141,9 @@ def collect_summary_statistics(data_dictionaries, half_bound, delta, steady_stat
 
 
 def convert_to_latex(summary_data, half_bound, delta, steady_state):
+    assert 'json_format_version' in summary_data and summary_data['json_format_version'] == JSON_VERSION_NUMBER, \
+        'Cannot process data from old JSON formats.'
+    del summary_data['json_format_version']
```

Comment:
Deleting the key seems pretty ugly (and likely to lead to confusion to one day). I assume this is because we want to keep the line `machine = summary_data.keys()[0]` as-is? Maybe there's a way to phrase that line that doesn't involve deleting the format key?